Explorer/Assets/DCL/Profiles/ProfileJsonDto.cs

```diff
@@ -0,0 +1,123 @@
+using CommunicationData.URLHelpers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace DCL.Profiles
+{
+    [Serializable]
+    public class EmoteJsonDto
+    {
+        public int slot;
+        public string urn;
+
+        public Emote ToEmote() =>
+            new ()
+            {
+                Slot = slot,
+                Urn = urn,
+            };
+    }
+
+    [Serializable]
+    public class AvatarColorJsonDto
+    {
+        public float r;
+        public float g;
+        public float b;
+        public float a;
+
+        public Color ToColor() =>
+            new (r, g, b, a);
+    }
+
+    [Serializable]
+    public class EyesJsonDto
+    {
+        public AvatarColorJsonDto color;
+    }
+
+    [Serializable]
+    public class HairJsonDto
+    {
+        public AvatarColorJsonDto color;
+    }
+
+    [Serializable]
+    public class SkinJsonDto
+    {
+        public AvatarColorJsonDto color;
+    }
+
+    [Serializable]
+    public class AvatarSnapshotJsonDto
+    {
+        public string face256;
+        public string body;
+    }
+
+    [Serializable]
+    public class AvatarJsonDto
+    {
+        public string bodyShape;
+        public string[] wearables;
+        public string[] forceRender;
+        public List<EmoteJsonDto> emotes;
+        public AvatarSnapshotJsonDto snapshots;
+        public EyesJsonDto eyes;
+        public HairJsonDto hair;
+        public SkinJsonDto skin;
+
+        public Avatar ToAvatar()
+        {
+            return new Avatar
+            {
+                Wearables = new HashSet<string>(wearables),
+                Emotes = emotes.ToDictionary(dto => dto.urn, dto => dto.ToEmote()),
+                EyesColor = eyes.color.ToColor(),
+                HairColor = hair.color.ToColor(),
+                SkinColor = skin.color.ToColor(),
+                FaceSnapshotUrl = URLAddress.FromString(snapshots.face256),
+                BodySnapshotUrl = URLAddress.FromString(snapshots.body),
+                BodyShape = bodyShape,
+                ForceRender = new HashSet<string>(forceRender),
+            };
+        }
+    }
+
+    [Serializable]
+    public class ProfileJsonDto
```

Comment:
As these DTOs are super heavy I expect significant GC pressure: it's worth pre-allocating/pooling them as well and deserialising into the existing instance:

When it comes to collections this is how it works with Unity's serialization:

>However, the behavior of JsonUtility.FromJsonOverwrite with arrays and lists can exhibit some inconsistencies. According to a report on Unity's Issue Tracker, when deserializing objects with array/list fields using this method, unexpected and inconsistent garbage collection (GC) allocations can occur. These include:
No allocation when deserializing into an array with the same number of elements.
New array allocated, with each element in the new array also allocated, when deserializing into an array with a different number of elements.
GC allocation call occurs when deserializing into a list with the same number of elements, but the list reference on the object doesn't change.
The list reference instance remains the same, but more allocations occur than in the case of an array with different numbers of elements, when deserializing into a list with different numbers of elements​​.

Reply:
Refactor into `OverwriteFromJsonAsync` passing a dto from the pool

Reply:
I've made the change, however including an empty constructor in the `Profile` forces miss-management on null fields